zc_install/sql/updates/mysql_upgrade_zencart_200.sql

```diff
@@ -40,6 +40,14 @@ TRUNCATE TABLE db_cache;
 
 ALTER TABLE products ADD products_mpn varchar(32) DEFAULT NULL AFTER products_model;
 
+# Product Dimensions fields
+ALTER TABLE products ADD products_length DECIMAL(8,4) DEFAULT NULL AFTER products_weight;
+ALTER TABLE products ADD products_width DECIMAL(8,4) DEFAULT NULL AFTER products_length;
+ALTER TABLE products ADD products_height DECIMAL(8,4) DEFAULT NULL AFTER products_width;
+ALTER TABLE products ADD product_ships_in_own_box TINYINT DEFAULT NULL AFTER products_height;
```

Comment:
Should the upgrade script also to an `ALTER TABLE products MODIFY` on the added fields so that the database schema matches the zc200 one, in case a site has installed an add-on that previously added these fields?

Reply:
Yes, definitely!

Reply:
Just to be sure, is ZC installer able to manage errors provoked by `ALTER ... ADD` if column already exists?
I was doing some testing and end with something like this to avoid MySQL errors:

```
SET @test_var_length = (SELECT NOT EXISTS(select * from information_schema.columns where table_schema = schema() and table_name = 'products' and column_name = 'products_length'));
SET @a = IF(@test_var_length = 1, 'ADD', 'MODIFY');
SET @st1 = CONCAT("ALTER TABLE products ", @a, " products_length DECIMAL(8,4) DEFAULT NULL AFTER products_weight;");
PREPARE stmt1 FROM @st1;
EXECUTE stmt1;
DEALLOCATE PREPARE stmt1;

SET @test_var_width = (SELECT NOT EXISTS(select * from information_schema.columns where table_schema = schema() and table_name = 'products' and column_name = 'products_width'));
SET @b = IF(@test_var_width = 1, 'ADD', 'MODIFY');
SET @st2 = CONCAT("ALTER TABLE products ", @b, " products_width DECIMAL(8,4) DEFAULT NULL AFTER products_length;");
PREPARE stmt2 FROM @st2;
EXECUTE stmt2;
DEALLOCATE PREPARE stmt2;

SET @test_var_height = (SELECT NOT EXISTS(select * from information_schema.columns where table_schema = schema() and table_name = 'products' and column_name = 'products_height'));
SET @c = IF(@test_var_height = 1, 'ADD', 'MODIFY');
SET @st3 = CONCAT("ALTER TABLE products ", @c, " products_height DECIMAL(8,4) DEFAULT NULL AFTER products_width;");
PREPARE stmt3 FROM @st3;
EXECUTE stmt3;
DEALLOCATE PREPARE stmt3;
```

Reply:
> Just to be sure, is ZC installer able to manage errors provoked by `ALTER ... ADD` if column already exists?

Yes. If those conditions are encountered, it silently logs a status note to the `upgrade_exceptions` table and carries on.
Same with the `Install SQL Patches` tool.

Reply:
I usually paste these sql mods in the Patch tool....I note this time, it outputs error messages:
![image](https://github.com/zencart/zencart/assets/4391026/a01e9876-7e03-409c-a6a7-617f8914bd87)

but the new columns **are** added correctly.